lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -111,16 +134,28 @@ def validate_presence_of(attr)
 
       # @private
       class ValidatePresenceOfMatcher < ValidationMatcher
+        def initialize(attribute)
+          super(attribute)
+          @options = {}
+        end
+
         def with_message(message)
           @expected_message = message if message
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @options[:allow_nil] = allow_nil
```

Comment:
While this makes more sense, I think currently in this file we use instance variables to store options, so let's do that.

Also, this qualifier doesn't tend to take an argument (`validate_inclusion_of` seems to be the only one that does this). Let's just leave it out for now, I might go through later and add in optional arguments for boolean qualifiers.